source/vulkan/vulkan_hooks.cpp

```diff
@@ -417,6 +650,7 @@ VkResult VKAPI_CALL vkCreateSwapchainKHR(VkDevice device, const VkSwapchainCreat
 	if (!runtime->on_init(*pSwapchain, *pCreateInfo, s_surface_windows.at(pCreateInfo->surface)))
 		LOG(ERROR) << "Failed to initialize Vulkan runtime environment on runtime " << runtime.get() << '.';
 
+	s_device_runtimes[device] = runtime;
```

Comment:
A bit questionable, considering that there may be multiple swapchains associated with a device and as such multiple runtimes. I realize the same issue exists in the other APIs and there we simply only update the first runtime, but that's not so great either.